Exercise_1.java

```diff
@@ -1,14 +1,31 @@
+// Time Complexity : o(log(n))
+// Space Complexity : o(log(n)) due to use of stack for recursion
+// Did this code successfully run on Leetcode :
+// Any problem you faced while coding this :
+
 class BinarySearch { 

```

Comment:
Nice try with recursion, try the iterative solution as well

Reply:
Thanks for the review. Added iterative binary search code